File: src/main/java/org/example/modelos/Propietario.java
package org.example.modelos;

import jakarta.persistence.*;

import java.util.List;

@Entity
@Table(name = "propietarios")
public class Propietario extends Usuario{
    @OneToMany(mappedBy = "propietario", cascade = CascadeType.ALL)
    private List<Piso> pisos;
    @ManyToMany(mappedBy = "propietarios")
    private List<Inquilino> inquilinos;     //Esta lista es la lista que recoge los inquilinos con los que habla el propietario
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "propietario_id")
    private List<Oferta> ofertas;

    public Propietario() {

    }

    public List<Inquilino> getInquilinos() {
        return inquilinos;
    }

    public void setInquilinos(List<Inquilino> inquilinos) {
        this.inquilinos = inquilinos;
    }

    public List<Piso> getPisos() {
        return pisos;
    }

    public void setPisos(List<Piso> pisos) {
        this.pisos = pisos;
    }
}

File: AndroidProyecto/app/src/main/java/com/example/androidappproyecto/modelos/Propietario.kt
package com.example.androidappproyecto.modelos

data class Propietario(
    val uid: String,
    val name: String,
    val email: String,
    val role: String,
    val phone: String,
    val address: String,

    val pisos: List<Piso>? = null,

    val inquilinos: List<InquilinoPropietario>? = null,

    val ofertas: List<Oferta>? = null,
    val contratos: List<Contrato>? = null
)